client/src/lobby/createGame.ts

```diff
@@ -262,14 +264,14 @@ const getVariant = (setting: string) => {
   return value;
 };
 
-const checkChanged = (settingName: string, value: boolean | string) => {
+const checkChanged = (settingName: keyof Settings, value: boolean | string) => {
   if (!misc.isKeyOf(settingName, globals.settings)) {
     throw new Error(`The setting of ${settingName} does not exist in the Settings class.`);
   }
 
   if (value !== globals.settings[settingName]) {
     // We must cast the settings to any since this assignment violates type safety
-    (globals.settings as any)[settingName] = value;
+    (globals.settings[settingName] as any) = value;
```

Comment:
The old one won't compile anymore, because there's no guarantee any has an indexer with the name. Now we typecheck before allowing assignment